### DISCUSSION TITLE
Can I stop new auto-layout nodes from appearing before they get put in the proper position?

### DISCUSSION THREAD

**Discussion by c45** [xyflow/xyflow#3465] [General]
Re this pro example:
https://pro.reactflow.dev/examples/auto-layout

I'm setting the default position of the new nodes to (0,0) and letting the auto-layout run to set the position, but there's a 1 second or so lag before the auto-layout runs which causes an annoying flicker effect of the edge moving across the screen.

**Comment by chrtze** [comment]
Hey @kyang8093, you can set the default style of the nodes and edges to `{ opacity: 0}`. It will be set to { opacity: 1 } by the layout hook to prevent the flickering.

Let me know if that works for you.

**Reply by c45** [reply]
Hi @chrtze , I've attempted this but am running into issues with a Custom node and edge type. It seems that using a custom type means that the CSS there is applied, but if I set the opacity to 0 like in the below, then the autolayout setting of opacity does not impact the custom edge. 

```export default function ThickerEdge(props: EdgeProps) {
  const { sourceX, sourceY, targetX, targetY, targetPosition, sourcePosition, id } = props;

  const [edgePath, labelX, labelY] = getBezierPath({
    sourceX,
    sourceY,
    sourcePosition,
    targetX,
    targetY,
    targetPosition,
  });

  return <BaseEdge {...props} path={edgePath} id={id} style={{ strokeWidth: 2, opacity: 0 }} />;
}
```

**Reply by chrtze** [reply]
If you are using a custom node or edge, you need to apply the style from the props, too:

```
  const { sourceX, sourceY, targetX, targetY, targetPosition, sourcePosition, id, style } = props;

  const [edgePath, labelX, labelY] = getBezierPath({
    sourceX,
    sourceY,
    sourcePosition,
    targetX,
    targetY,
    targetPosition,
  });

  return <BaseEdge {...props} path={edgePath} id={id} style={{ strokeWidth: 2, opacity: 0, ...style }} />;
```

**Reply by c45** [reply]
Thank you @chrtze  -- this worked! Very happy about the level of support y'all provided, you rock!
